src/jdk.compiler/share/classes/com/sun/tools/javac/parser/JavacParser.java

```diff
@@ -4020,7 +4020,7 @@ protected JCClassDecl recordDeclaration(JCModifiers mods, Comment dc) {
                     for (JCVariableDecl param : headerFields) {
                         tmpParams.add(F.at(param)
                                 // we will get flags plus annotations from the record component
-                                .VarDef(F.Modifiers(Flags.PARAMETER | Flags.GENERATED_MEMBER | param.mods.flags & Flags.VARARGS,
+                                .VarDef(F.Modifiers(Flags.PARAMETER | Flags.GENERATED_MEMBER | Flags.MANDATED | param.mods.flags & Flags.VARARGS,
```

Comment:
Should the use of MANDATED be conditional on the target class file version?

Reply:
I advise also writing a core reflection test that uses access flags (java.lang.reflect.AccessFlag).

Reply:
> Should the use of MANDATED be conditional on the target class file version?

I don't think this is needed. The JLS mentions the parameters of canonical record constructors in Java 14 already. Previous versions don't have records and shouldn't enter that method at all.

> I advise also writing a core reflection test that uses access flags (java.lang.reflect.AccessFlag).

Will do. I assume the `jdk/java/lang/reflect/AccessFlag` folder is a good place for that?

Reply:
> > Should the use of MANDATED be conditional on the target class file version?
> 
> I don't think this is needed. The JLS mentions the parameters of canonical record constructors in Java 14 already. Previous versions don't have records and shouldn't enter that method at all.

Agreed; I double-checked and the ACC_MANDATED flag was defined in that location since before JDK 14's format.

> 
> > I advise also writing a core reflection test that uses access flags (java.lang.reflect.AccessFlag).
> 
> Will do. I assume the `jdk/java/lang/reflect/AccessFlag` folder is a good place for that?

Yes; that an appropriate  home for such tests.